examples/bot-grnu/README.md

```diff
@@ -0,0 +1,75 @@
+# GCP Release Notes Updater BOT
+
+This solution was built with the purpose of demonstrating how to develop a fully serverless
+Hangouts Chat BOT. The purpose of the solution is to daily update Release Notes of selected GCP products
+and notify users in case of news. This document will guide you through the necessary steps to set up
+the entire solution on Google Cloud Platform (GCP).
+
+## Architecture
+
+The image below contains a high level diagram of the solution.
+
+![](img/architecture.png)
+
+Main features:
+* daily synchronization of all registered feeds
+* ability to manually query the BOT for getting latest release notes about a specific solution (e.g. bigquery)
+* automatic message push in registered rooms
+  * one dedicated thread per solution
+  * all subsequents solution updates displayed on the same thread
+
+## Prerequisites
+
+* G Suite domain where to connect the Hangouts Chat
+* GCP project where to deploy the application referenced with
+  * **[PROJECT_ID]**
+  * **[PROJECT_NUMBER]**
+* Enabled [Google Hangouts Chat API](https://console.cloud.google.com/apis/library/chat.googleapis.com) in the selected GCP project
+* A newly generated [UUID](https://www.uuidgenerator.net/) code referenced with
+  * **[BOT_UUID]**
+* Configuration completed for the [Google Hangouts Chat API](https://console.cloud.google.com/apis/library/chat.googleapis.com) 
+  * _Bot Name_ : the name of the BOT that will appear on Hangouts Chat
+  * _Avatar URL_: the url of the BOT avatar
+  * _Description_: a desciption of the BOT
+  * Enable both "_Bot works in direct messages_" and "_Bot works in rooms_" flags
+  * _Connections String_: select "Bot URL"
+  * _Bot URL_: https://[PROJECT_ID].appspot.com/rest/chat/bot/listen/[BOT_UUID]
+  * _Permissions_ : decide what you prefer. People you will add here will have visibility of the BOT
+* A _Service Account_ 
+* _Service Account JSON key_ saved in the following file: `src/main/webapp/SA.json`
```

Comment:
This might be a dumb question, but why do we need a service account at all? This readme doesn't say what permissions to grant it access to. I couldn't see where the key file is accessed, is it loaded implicitly?

Reply:
It is required to enable BOT to send messages (reference [here](https://developers.google.com/hangouts/chat/how-tos/service-accounts)).
What is the best practice here? I tried using the AppEngine default credential but unfortunately it does not work: a service account is required.